Source/Tools/FEXLoader/FEXLoader.cpp

```diff
@@ -194,15 +193,53 @@ void RootFSRedirect(fextl::string* Filename, const fextl::string& RootFS) {
   }
 }
 
-bool RanAsInterpreter(const char* Program) {
-  return ExecutedWithFD || FEXLOADER_AS_INTERPRETER;
+FEX::Config::PortableInformation ReadPortabilityInformation() {
+  const FEX::Config::PortableInformation BadResult {false, {}};
+  const char* PortableConfig = getenv("FEX_PORTABLE");
+  if (!PortableConfig) {
+    return BadResult;
+  }
+
+  uint32_t Value {};
+  std::string_view PortableView {PortableConfig};
+
+  if (std::from_chars(PortableView.data(), PortableView.data() + PortableView.size(), Value).ec != std::errc {} || Value == 0) {
+    return BadResult;
+  }
+
+  // Read the FEXInterpreter path from `/proc/self/exe` which is always a symlink to the absolute path of the executable running.
+  // This way we can get the parent path that the application is executing from.
+  char SelfPath[PATH_MAX];
+  auto Result = readlink("/proc/self/exe", SelfPath, PATH_MAX);
+  if (Result == -1) {
+    return BadResult;
+  }
+
+  std::string_view SelfPathView {SelfPath, std::min<size_t>(PATH_MAX, Result)};
+
+  // Extract the absolute path from the FEXInterpreter path
+  return {true, fextl::string {SelfPathView.substr(0, SelfPathView.find_last_of('/') + 1)}};
 }
 
-bool IsInterpreterInstalled() {
-  // The interpreter is installed if both the binfmt_misc handlers are available
-  // Or if we were originally executed with FD. Which means the interpreter is installed
+bool RanAsInterpreter(bool ExecutedWithFD) {
+  return ExecutedWithFD || FEXLOADER_AS_INTERPRETER;
+}
 
-  return ExecutedWithFD || (access("/proc/sys/fs/binfmt_misc/FEX-x86", F_OK) == 0 && access("/proc/sys/fs/binfmt_misc/FEX-x86_64", F_OK) == 0);
+/**
+ * @brief Queries if FEX is installed as a binfmt_misc interpreter
+ *
+ * @param ExecutedWithFD If FEXInterpreter was executed using a binfmt_misc FD handle from the kernel
+ * @param Portable Portability information about FEX being run in portable mode
+ *
+ * @return true if the binfmt_misc handlers are installed and being used
+ */
+bool QueryInterpreterInstalled(const bool ExecutedWithFD, const FEX::Config::PortableInformation& Portable) {
```

Comment:
`const` on value parameters has no meaningful effect for short functions. It's only useful to avoid accidental modification in the implementation of large functions, but it's ignored externally either way.

```suggestion
bool QueryInterpreterInstalled(bool ExecutedWithFD, const FEX::Config::PortableInformation& Portable) {
```